### PR TITLE
Search screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 75823: Search screen](https://edu-redmine.sun-asterisk.vn/issues/75823)
## WHAT
- Search screen 
- Recent search
- result/no result
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/HN_Android_2024_Travenor_App/assets/165371062/9f7b569d-ffbf-45ac-9b61-e086a04f042a

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/travenor/screen/search/adapter/SearchResultAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    override fun getItemCount(): Int {
+        return if (mSearchResultList.isEmpty()) 1 else mSearchResultList.size
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        if (getItemViewType(position) == TYPE_RESULT) {
+            val place = mSearchResultList[position]
+            val binding = (holder as ResultViewHolder).binding
+
```
Comment (ghost) [inline app/src/main/java/com/example/travenor/screen/search/adapter/SearchResultAdapter.kt:72]: viết cái hàm trong ResultViewHolder cho nó gọn e

**Review by ghost — APPROVED** [review]

File: app/src/main/java/com/example/travenor/screen/search/SearchFragment.kt [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    private val mRecentSearchAdapter: RecentSearchAdapter by lazy { RecentSearchAdapter() }
+    private val mSearchResultAdapter: SearchResultAdapter by lazy { SearchResultAdapter() }
+    private var mPresenter: SearchPresenter? = null
+
+    override fun inflateViewBinding(inflater: LayoutInflater): FragmentSearchBinding {
+        return FragmentSearchBinding.inflate(inflater)
+    }
+
+    private fun setLayoutBelowSystemBar() {
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/screen/search/SearchFragment.kt]: Các private func em move vào cuối class nhé,
Thống nhất cho dễ đọc code sau này!

sx ntn:
- public method
- Override func of parent class
- Override func interface implement
- private func
- nested/inner class
- compation object ..

File: app/src/main/java/com/example/travenor/screen/search/adapter/SearchResultAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    override fun getItemCount(): Int {
+        return if (mSearchResultList.isEmpty()) 1 else mSearchResultList.size
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        if (getItemViewType(position) == TYPE_RESULT) {
+            val place = mSearchResultList[position]
+            val binding = (holder as ResultViewHolder).binding
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/screen/search/adapter/SearchResultAdapter.kt]: không nên force cast như này!

**Review by daolq-2712 — CHANGES_REQUESTED** [review]
Pls check my comments!

File: app/src/main/java/com/example/travenor/screen/search/SearchFragment.kt (hunk 2 above)
Comment (ghost) [reply app/src/main/java/com/example/travenor/screen/search/SearchFragment.kt]: Vâng a

**Review by daolq-2712 — APPROVED** [review]
LGTM!

**Pull request merged**